### DISCUSSION TITLE
For loop JSX destructuring

### DISCUSSION THREAD

**Discussion by ivanjeremic** [solidjs-community/solid-dsl#12] [Proposals]
I think there needs to be something like JSX destructuring and loops would look cleaner without a function.

**Now:**
```ts
<For each={items()}>
   {(item, index) => <div>id: {item.id} - index: {index}</div>}
</For>
```

**After:**
```ts
// parentheses
<For each={items()} (item, index)>
  <div>id: {item.id} - index: {index}</div>
</For>

// or curly brackets
<For each={items()} {item, index}>
  <div>id: {item.id} - index: {index}</div>
</For>
```

**Comment by lxsmnsyc** [comment]
Currently in conflict with https://github.com/solidjs-community/solid-dsl/discussions/9

We can probably redesign this in a way that it doesn't have to follow the markup structure.

**Comment by Jacob-Lockwood** [comment]
Maybe this syntax with `=>` would be more familiar?

```jsx
<For each={items()} (item, index) =>
  <div>id: {item.id} - index: {index}</div>
</For>
```

It still has the feeling of closing the opener block because `=>` is both an arrow and ends with `>`
